ecosystem/indexer/src/indexer/fetcher.rs

```diff
@@ -19,21 +25,66 @@ impl TransactionFetcher {
         Self {
             client,
             version: starting_version.unwrap_or(0),
+            transactions_buffer: Default::default(),
         }
     }
 
     pub fn set_version(&mut self, version: u64) {
         self.version = version;
     }
 
+    /// Fetches the next version based on its internal version counter
+    /// Under the hood, it fetches 100 (when needed), and uses that buffer to feed out
+    /// In the event it can't, it will keep retrying every 5s
     pub async fn fetch_next(&mut self) -> Transaction {
-        let transaction = self.fetch_version(self.version).await;
+        let mut transactions_buffer = self.transactions_buffer.lock().await;
+        if transactions_buffer.is_empty() {
```

Comment:
not super clear why we need the mutex but we don't need atomics on the version -- is this some early code for parallelism?

Reply:
version is only ever incremented within the context of a mutex